SDK/Utils/Extensions/StringExtension.swift

```diff
@@ -14,7 +14,8 @@ extension String {
     func webcalURL() -> URL? {
         let urlBegin = self.range(of: "://")!.upperBound
         let urlString = self[urlBegin..<self.endIndex]
-        let webcal = URL(string: "webcal://\(urlString)")
+        let urlU = "\(urlString)&u=\(Config.uuid)"
```

Comment:
isn't the u parameter already in the API response now (after the API query also got the parameter)?

Reply:
But we don't have it for the calendar url, that's why it's used here. it is the same uuid sent to the api

Reply:
IRRC, the `u` param is appended automatically to the calendar urls when you also add it to the API call, so please check that first.
Also the way you append it, presumes there already are other parameters which may not necessarily true for every calendar URL returned by the API.
In general I'd prefer using URL modification methods than String handling.

Reply:
@dmfs the url we get from the API is like this: https://holidays.schedjoules.com/calendars/766e8a162f82?l=en , that's why the `u` parameter needs to be added.

I made a change to use URLComponents instead of string.


I didn't use the URL methods to append components in swift, since don't give us what we need since they add the user id before the other query items so we end up with urls like this:
https://holidays.schedjoules.com/calendars/766e8a162f82/u=xxxxxxxx?l=en or 
https://holidays.schedjoules.com/calendars/766e8a162f82/u=xxxxxxxx/?l=en